data/ledger_test.go

```diff
@@ -629,6 +632,32 @@ func TestLedgerErrorValidate(t *testing.T) {
 			case <-expectedMessages:
 				// only debug messages should be reported
 			case um := <-unexpectedMessages:
+				if strings.Contains(um, "before dbRound") {
```

Comment:
> round 0 before dbRound 2
Handled this as well. This is from Lookup path that is part of block evaluation. The first one handled came from EnsureBlock and the second from AddBlock but because of the same reason - while `AddBlock` goroutine was suspended, the `EnsureValidatedBlock` add 6+ blocks so that ledger committed account changed and trackers advanced.

Reply:
Now we uncover another error...
```
    ledger_test.go:677:
        	Error Trace:	/Users/shantkarakashian/go/src/github.com/algorand/go-algorand/data/ledger_test.go:677
        	Error:      	Should be empty, but was failed to delete hash '000001830087367e5228afa552f1eaf821f979a8352e527bb1e8119b761ae4598d5a5bf8' from merkle trie for account 7777777777777777777777777777777777777777777777777774MSJUVU
        	Test:       	TestLedgerErrorValidate
        	Messages:   	failed to delete hash '000001830087367e5228afa552f1eaf821f979a8352e527bb1e8119b761ae4598d5a5bf8' from merkle trie for account 7777777777777777777777777777777777777777777777777774MSJUVU
```

coming from catchpointtracker.go:
```
 952// accountsUpdateBalances applies the given compactAccountDeltas to the merkle trie
 953func (ct *catchpointTracker) accountsUpdateBalances(accountsDeltas compactAccountDeltas, resourcesDeltas compactResourcesDeltas, kvDeltas map[string]modifiedKvV$
 954        if !ct.catchpointEnabled() {
 955                return nil
 956        }
 957        accumulatedChanges := 0
 958
 959        for i := 0; i < accountsDeltas.len(); i++ {
 960                delta := accountsDeltas.getByIdx(i)
 961                if !delta.oldAcct.AccountData.IsEmpty() {
 962                        deleteHash := trackerdb.AccountHashBuilderV6(delta.address, &delta.oldAcct.AccountData, protocol.Encode(&delta.oldAcct.AccountData))
 963                        deleted, err := ct.balancesTrie.Delete(deleteHash)
 964                        if err != nil {
 965                                return fmt.Errorf("failed to delete hash '%s' from merkle trie for account %v: %w", hex.EncodeToString(deleteHash), delta.addres$
 966                        }
 967                        if !deleted {
 968                                ct.log.Errorf("failed to delete hash '%s' from merkle trie for account %v", hex.EncodeToString(deleteHash), delta.address)
 969                        } else {
```